Sources/Foundation/NSError.swift

```diff
@@ -891,8 +891,8 @@ func _convertNSErrorToError(_ error: NSError?) -> Error {
 
 public // COMPILER_INTRINSIC
 func _convertErrorToNSError(_ error: Error) -> NSError {
-    if let object = _extractDynamicValue(error as Any) {
-        return unsafeDowncast(object, to: NSError.self)
+    if let object = _extractDynamicValue(error as Any), let asNS = object as? NSError {
```

Comment:
I suppose CI will tell us, but I don't expect this will work here. I don't know the full details, but I believe that `_convertErrorToNSError` _is_ effectively the implementation of `as NSError`, so I suspect this is likely to cause infinite recursion. To check that, I compiled

```swift
public func foo(_ e: any Error) -> NSError {
    e as NSError
}
``

in GodBolt on Linux and saw that it indeed compiles down to a call to this function:

```
output.foo(Swift.Error) -> Foundation.NSError:
        b       ($s10Foundation22_convertErrorToNSErroryAA0E0Cs0C0_pF)
```

That said, I don't know the answer to what this should be instead. I'd be interested to know what the type of `object` is here in the failing case

Reply:
I did run CI on this locally, but it's possible we for some reason do not hit this path.